app/src/main/java/de/grobox/transportr/utils/IntentUtils.kt

```diff
@@ -100,20 +103,16 @@ object IntentUtils {
             flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
             data = geo
         }
-        val intents = context.packageManager.queryIntentActivities(intent, 0).mapNotNull {
-            val packageName = it.activityInfo.packageName
-            if (context.packageName != packageName)
-                Intent(intent).apply { `package` = packageName }
-            else null
-        }
-        if (intents.isEmpty()) {
+        val intentChooser = Intent.createChooser(intent,  context.getString(R.string.show_location_in))
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
+            // exclude Transportr from list on Android >= 7
+            intentChooser.putExtra(Intent.EXTRA_EXCLUDE_COMPONENTS, arrayOf(ComponentName(context, MapActivity::class.java)))
```

Comment:
filtering out Transportr from the intent chooser now only works on Android 7+. On older versions, we have the small regression that Transportr is shown again as an option in the chooser (same as before #713), but it simplifies the code a lot.